### PR TITLE
refactor(cdu/fms): code cleanup (part 3)

### PR DESCRIPTION
**Pull request by MisterChocker**
## Summary of Changes

- Added tasks (TODOs) (for future code refactoring (part4 or 5))
- Refactored fms managed speed
- Separated MCDU and FMS code

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Do patterns, do normal flight, test the mcdu functions and try to brake the system ^^.
Also observer managed speeds, flight phases as well as flight phase and mode changes.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js [hunk 1]
```diff
@@ -1651,18 +1651,6 @@ class FMCMainDisplay extends BaseAirliners {
...
-    //TODO: MCDU action
-    //TODO: figure out usage of takeOffTrim
-    updateTakeOffTrim() {
-        let d = (this.zeroFuelWeightMassCenter - 13) / (33 - 13);
-        d = Math.min(Math.max(d, -0.5), 1);
-        let dW = (this.getWeight(true) - 400) / (800 - 400);
-        dW = Math.min(Math.max(dW, 0), 1);
-        const minTrim = 3.5 * dW + 1.5 * (1 - dW);
-        const maxTrim = 8.6 * dW + 4.3 * (1 - dW);
-        this.takeOffTrim = minTrim * d + maxTrim * (1 - d);
```
Comment (MisterChocker) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js:1655]: This was added in the initial import and had no real usage.

**Comment by Lazarus320** [comment]
Category: QA Tester 
Name: Lazarus#4286 
Date of testing: 29/1/2021
Version of the sim: 1.12.13.0
PR Tested: #3182 
Tier of testing: 2 
Changes to observe:
CDU/FMS refactor.

Testing technique: DId few patterns and a normal flights 
Steps you've made to test the Pull Request:
1. Tested  managed speed, flight phases mode changes.
2. Tested MCDU functions. 
 
Flight notes: N/A

Media: N/A

Issues (If Any): N/A
 
Overall Rating: Good
Conclusions of the testing: Everything was working as usual with no issue.

**Review by Lucky38i — APPROVED** [review]
LGTM, idk why the methods were awaited anyway.

**Comment by FlightSimmer7700** [comment]
Category: QA Tester
Name: FlightSimmer7700#3468
Date of testing: 30/1/21
Version of the sim: 1.12.13.0
PR Tested: #3182 
Tier of testing: 2
Changes to observe:
MCDU

Testing technique: Did a full flight
Steps you've made to test the Pull Request:
1. Did a full flight
2. Used simbrief uplink
3. Went Direct


Overall Rating: Good
Conclusions of the testing: Did a full flight and found/had no issues

**Review by tracernz — COMMENTED** [review]
Re: ignored promises in 15b19f56523db7ef2afcfc5b97af8f0437800274
I'm fine with this change, but for my own benefit I'd like to clarify if this is actually a problem? I'm guessing the result would be that the called function doesn't get to run until the event loop runs for some other reason? I don't know how the JS event loop works so my mental model is based on other languages/frameworks, especially Qt.

**Review by tracernz — APPROVED** [review]

**Comment by davidwalschots** [comment]
@tracernz No. That change is a bit unusual. A `then` without any content is equal to not adding `then`. I'd revert that to no longer use `then`.

> I'm guessing the result would be that the called function doesn't get to run until the event loop runs for some other reason?

This is false. The `SetSimVarValue` function is executed immediately and at some point returns a `Promise`. The `Promise` can be used to continue execution after `SetSimVarValue` indicates it has completed, with the result `SetSimVarValue` provides (if any). This is only relevant if you depend on something `SetSimVarValue` executed in the code that follows the call to `SetSimVarValue`.

What will happen in this code is: 
1. `SetSimVarValue` executes and returns a `Promise`.
2. Any remaining code left to be executed outside of `SetSimVarValue` is executed.
3. `SetSimVarValue` resolves the `Promise` (means it says: "hey I'm done!").
4. At some point in time the work in `then(function() { console.log("I'm executed much later!") })` gets executed.

**Comment by sidnov** [comment]
Category: QA Tester
Name: sidnov#8337
Date of testing: 30/01/2021
Version of the sim: 1.12.13.0
PR Tested: #3182 
Tier of testing: 1
Changes to observe: Check CDU/FMS bugs if any

Testing technique: Tryout

### Steps you've made to test the Pull Request:

1. Spawned C/D at WSSS. Prepared for a normal IFR flight from WSSS to VTBS.
2. Filled MCDU pages normally during preflight.
3. Input route using airways, and did not specify waypoints. Intersection was detected automatically
4. Also tested ATSU features, printer etc. midflight.
5. Used a STAR for approach into VTBS. Cancelled STAR midway and proceeded on vectors.
6. Used ILS for RWY 01R
7. Landed normally.

### Flight notes (When done a full flight): 
Route: WSSS SID VMR B469 VPK M751 VKB M644 TONIK M904 BUT STAR VTBS
Weather Engine: REX 6
Network: POSCON

### Issues (If Any): None

### Overall Rating: Good

Conclusions of the testing: The flight went as per plan, and no abnormal behavior was observed.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js [hunk 2]
```diff
@@ -2949,65 +2957,79 @@ class FMCMainDisplay extends BaseAirliners {
...
-                    this.addNewMessage(NXSystemMessages.formatError);
-                    return false;
-                }
+        if (s === FMCMainDisplay.clrValue) {
+            this.flaps = NaN;
+            this.ths = NaN;
+            SimVar.SetSimVarValue("L:A32NX_TO_CONFIG_FLAPS", "number", 0);
+            SimVar.SetSimVarValue("L:A32NX_TO_CONFIG_FLAPS_ENTERED", "bool", false);
+            SimVar.SetSimVarValue("L:A32NX_TO_CONFIG_THS", "degree", 0);
+            SimVar.SetSimVarValue("L:A32NX_TO_CONFIG_FLAPS_ENTERED", "bool", false);
```
Comment (tracernz) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js:2966]: This line ended up twice (see 2 lines up). Not sure if it’s from master or merge conflict fix-up?

Comment (MisterChocker) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js:2966]: Looks like it was like this before.

image

**Comment by cianwhite** [comment]
Category: QA Tester
Name: Cian#3596
Date of testing: 29/1/2021
Version of the sim: 1.12.13.0
PR Tested: #3182
Tier of testing: 2
Changes to observe: CDU/FMS refactor.

Testing technique: Short flight and spawning in on approach
Steps you've made to test the Pull Request:

1. Spawned in at EGCC, started as normal and loaded F-PLAN from simbrief
2. Took off and climbed to cruise as normal
3. Descended as usual and landed on ILS 26L at EGKK as normal
4. Spawned in on an approach for EGKK 26L
5. Landed as normal
Flight notes: LIST2Z LISTO L612 HON N859 KIDLI KIDL1G


Issues (If Any): None

Overall Rating: Good
Conclusions of the testing: No issues were observed by the CDU/FMS cleanup, everything worked fine.

**Pull request merged**